core/experiments/manager.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+namespace Elementor\Core\experiments;
+
+use Elementor\Core\Base\Base_Object;
+use Elementor\Tools;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+class Manager extends Base_Object {
+
+	const STATUS_ALPHA = 1;
+
+	const STATUS_BETA = 2;
+
+	const STATE_DEFAULT = 0;
+
+	const STATE_ACTIVE = 1;
+
+	const STATE_INACTIVE = 2;
+
+	private $features;
+
+	/**
+	 * Add Feature
+	 *
+	 * @since 3.1.0
+	 * @access public
+	 *
+	 * @param array $options {
+	 *     @type string $name
+	 *     @type string $title
+	 *     @type string $description
+	 *     @type string $status
+	 *     @type string $default
+	 * }
+	 *
+	 * @return array|null
+	 */
+	public function add_feature( array $options ) {
+		if ( isset( $this->features[ $options['name'] ] ) ) {
+			return null;
+		}
+
+		$allowed_options = [ 'name', 'title', 'description', 'status', 'default' ];
+
+		$experimental_data = array_intersect_key( $options, array_flip( $allowed_options ) );
+
+		$default_experimental_data = [
+			'description' => '',
+			'status' => self::STATUS_ALPHA,
+			'default' => self::STATE_INACTIVE,
+		];
+
+		$experimental_data = array_merge( $default_experimental_data, $experimental_data );
+
+		$state = $this->get_saved_feature_state( $options['name'] );
+
+		if ( ! $state ) {
+			$state = self::STATE_DEFAULT;
+		}
+
+		$experimental_data['state'] = $state;
+
+		$this->features[ $options['name'] ] = $experimental_data;
+
+		return $experimental_data;
+	}
+
+	/**
+	 * Remove Feature
+	 *
+	 * @since 3.1.0
+	 * @access public
+	 *
+	 * @param string $feature_name
+	 */
+	public function remove_feature( $feature_name ) {
+		if ( isset( $this->features[ $feature_name ] ) ) {
+			unset( $this->features[ $feature_name ] );
+		}
+	}
+
+	/**
+	 * Get Features
+	 *
+	 * @since 3.1.0
+	 * @access public
+	 *
+	 * @param string $feature_name Optional. Default is null
+	 *
+	 * @return array|null
+	 */
+	public function get_features( $feature_name = null ) {
+		return self::get_items( $this->features, $feature_name );
+	}
+
+	/**
+	 * Is Feature Active
+	 *
+	 * @since 3.1.0
+	 * @access public
+	 *
+	 * @param string $feature_name
+	 *
+	 * @return bool
+	 */
+	public function is_feature_active( $feature_name ) {
+		$feature = $this->get_features( $feature_name );
+
+		if ( ! $feature || self::STATE_INACTIVE === $feature['state'] ) {
+			return false;
+		}
+
+		if ( self::STATE_DEFAULT === $feature['state'] ) {
+			return self::STATE_ACTIVE === $feature['default'];
+		}
+
+		return true;
+	}
+
+	/**
+	 * Set Feature Default State
+	 *
+	 * @since 3.1.0
+	 * @access public
+	 *
+	 * @param string $feature_name
+	 * @param int $default_state
+	 */
+	public function set_feature_default_state( $feature_name, $default_state ) {
```

Comment:
How can it be used before the features are registered?

Reply:
It can't.